### DISCUSSION TITLE
[Fixed] 'OCI runtime create failed' error on macOS

### DISCUSSION THREAD

**Discussion by kienstra** [kienstra/localci#121] [General]
You might see this error:

>Error: 
Unexpected environment preparation error: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:385: applying cgroup configuration for process caused: cannot enter cgroupv2 "/sys/fs/cgroup/docker/17a27748567cb754976fca9c8c2366d65bc6235308785641092932024931bf17/e6d40747abd04130414779b311791222067ab1913f7588d90f803c2ced3abb65" with domain controllers -- it is in an invalid state: unknown
Step failed
Task failed

Thanks to [Matt](https://github.com/CircleCI-Public/circleci-cli/issues/676#issue-1173135216) and [Mason](https://github.com/CircleCI-Public/circleci-cli/issues/676#issuecomment-1100434492), running this in your local should fix it:
```sh
rm ~/.circleci/build_agent_settings.json
echo '{"LatestSha256":"sha256:008ba7f4223f1e26c11df9575283491b620074fa96da6961e0dcde47fb757014"}' > ~/.circleci/build_agent_settings.json
```

This sets the Docker image that will run the CircleCI® job. That image shouldn't throw the error.

**Comment by aronchick** [comment]
I've done this several times (M1 mac, fresh install) and same problem.

**Comment by aronchick** [comment]
```
❯ more ~/.circleci/build_agent_settings.json
{"LatestSha256":"sha256:008ba7f4223f1e26c11df9575283491b620074fa96da6961e0dcde47fb757014"}
```

**Comment by kienstra** [comment]
Hi @aronchick,
Thanks for letting me know about this, looking at it now.

**Comment by kienstra** [comment]
Hi @aronchick,
Could you try only doing:

```sh
rm ~/.circleci/build_agent_settings.json
```

…without doing:

```sh
echo '{"LatestSha256":"sha256:008ba7f4223f1e26c11df9575283491b620074fa96da6961e0dcde47fb757014"}' > ~/.circleci/build_agent_settings.json
```

That `echo` command ideally won't be needed on an M1 Mac, but maybe it is. I'll help debug if it is needed.
